src/main/kotlin/uk/gov/justice/digital/hmpps/courtregister/resource/CourtResource.kt

```diff
@@ -3,63 +3,103 @@ package uk.gov.justice.digital.hmpps.courtregister.resource
+import javax.validation.constraints.NotBlank
 import javax.validation.constraints.Size
 
 @RestController
 @Validated
 @RequestMapping("/courts", produces = [MediaType.APPLICATION_JSON_VALUE])
 class CourtResource(private val courtService: CourtService) {
   @GetMapping("/id/{courtId}")
-  @Operation(summary = "Get specified court", description = "Information on a specific court")
-  @ApiResponses(
-    value = [
+  @Operation(
+    summary = "Get specified court",
+    description = "Information on a specific court",
+    responses = [
       ApiResponse(
         responseCode = "200",
-        description = "Successful Operation",
+        description = "Court Information Returned",
         content = [Content(mediaType = "application/json", schema = Schema(implementation = CourtDto::class))]
+      ),
+      ApiResponse(
+        responseCode = "400",
+        description = "Incorrect request to get court information",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))]
+      ),
+      ApiResponse(
+        responseCode = "404",
+        description = "Court ID not found",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))]
       )
     ]
   )
   fun getCourtFromId(
     @Schema(description = "Court ID", example = "ACCRYC", required = true)
-    @PathVariable @Size(max = 12, min = 2) courtId: String
+    @PathVariable @Size(max = 12, min = 2, message = "Court ID must be between 2 and 12") courtId: String
   ): CourtDto =
     courtService.findById(courtId)
 
   @GetMapping("")
-  @Operation(summary = "Get all courts", description = "All courts")
-  @ApiResponses(
-    value = [
+  @Operation(
+    summary = "Get all active courts",
+    description = "All courts (active only)",
+    responses = [
       ApiResponse(
         responseCode = "200",
-        description = "Successful Operation",
-        content = [Content(mediaType = "application/json")]
+        description = "All Court Information Returned",
+        content = arrayOf(
+          Content(
+            mediaType = "application/json",
+            array = ArraySchema(schema = Schema(implementation = CourtDto::class))
+          )
+        )
       )
     ]
   )
   fun getCourts(): List<CourtDto> =
-    courtService.findAll()
+    courtService.findAll(true)
+
+  @GetMapping("/all")
+  @Operation(
+    summary = "Get all active and inactive courts",
+    description = "All active/inactive courts",
+    responses = [
+      ApiResponse(
+        responseCode = "200",
+        description = "All Court Information Returned (Active only)",
+        content = arrayOf(
+          Content(
+            mediaType = "application/json",
+            array = ArraySchema(schema = Schema(implementation = CourtDto::class))
+          )
+        )
+      )
+    ]
+  )
+  fun getAllCourts(): List<CourtDto> =
```

Comment:
Maybe better with `getActiveCourts` and `getAllCourts`? `getCourts` and `getAllCourts` sounds the same to me.